website/versioned_docs/version-v1.3_alpha/troubleshooting.md

```diff
@@ -49,3 +49,113 @@ You can generate the missing `.dat` files using the following command:
+    at async Environment.run (/home/user/Documents/maci/node_modules/.pnpm/hardhat@2.22.2_ts-node@10.9.2_typescript@5.4.3/node_modules/hardhat/src/internal/core/runtime-environment.ts:192:14)
+    at async main (/home/user/Documents/maci/node_modules/.pnpm/hardhat@2.22.2_ts-node@10.9.2_typescript@5.4.3/node_modules/hardhat/src/internal/cli/cli.ts:323:7) {
+  generatedMessage: true,
+  code: 'ERR_ASSERTION',
+  actual: false,
+  expected: true,
+  operator: '=='
+}
+```
+
+### Cannot filter non-indexed parameters
+
+If you encountered the following error log:
+
+```
+TypeError: cannot filter non-indexed parameters; must be null (argument="contract._messageRoot", value=4658669366154318429589992378027343879410506110087584229116806667655891474709, code=INVALID_ARGUMENT, version=6.11.1)
+    at makeError (/home/user/Documents/maci/node_modules/.pnpm/ethers@6.11.1/node_modules/ethers/src.ts/utils/errors.ts:687:21)
+    at assert (/home/user/Documents/maci/node_modules/.pnpm/ethers@6.11.1/node_modules/ethers/src.ts/utils/errors.ts:715:25)
+    at assertArgument (/home/user/Documents/maci/node_modules/.pnpm/ethers@6.11.1/node_modules/ethers/src.ts/utils/errors.ts:727:5)
+    at /home/user/Documents/maci/node_modules/.pnpm/ethers@6.11.1/node_modules/ethers/src.ts/abi/interface.ts:1047:31
+    at Array.forEach (<anonymous>)
+    at Interface.encodeFilterTopics (/home/user/Documents/maci/node_modules/.pnpm/ethers@6.11.1/node_modules/ethers/src.ts/abi/interface.ts:1042:16)
+    at /home/user/Documents/maci/node_modules/.pnpm/ethers@6.11.1/node_modules/ethers/src.ts/contract/contract.ts:108:39
+    at processTicksAndRejections (node:internal/process/task_queues:95:5)
+    at async getSubInfo (/home/user/Documents/maci/node_modules/.pnpm/ethers@6.11.1/node_modules/ethers/src.ts/contract/contract.ts:502:18)
+    at async Proxy.queryFilter (/home/user/Documents/maci/node_modules/.pnpm/ethers@6.11.1/node_modules/ethers/src.ts/contract/contract.ts:938:38) {
+  code: 'INVALID_ARGUMENT',
+  argument: 'contract._messageRoot',
+  value: 4658669366154318429589992378027343879410506110087584229116806667655891474709n,
+  shortMessage: 'cannot filter non-indexed parameters; must be null'
+}
+```
+
+This could happen during running `genProofs` in `cli` package, or running `prove` in `contracts` package.
+Be aware that we updated several parameters to `indexed`:
+
+```javascript
+event MergeMaciStateAqSubRoots(uint256 indexed _numSrQueueOps);
+event MergeMaciStateAq(uint256 indexed _stateRoot, uint256 indexed _numSignups);
+event MergeMessageAqSubRoots(uint256 indexed _numSrQueueOps);
+event MergeMessageAq(uint256 indexed _messageRoot);
+```
+
+Please remember to pull the latest MACI repo updates(`git fetch origin && git pull origin dev`) and run `build` in the `contracts` package.
```

Comment:
I think it might be better to suggest to run `build` in the monorepo roop, vs only in the contracts folder

Reply:
well, actually should run `build` both in the monorepo and the contracts folder, since I found even I ran `build` in the monorepo, the solidity files are not recompiled.